### ISSUE TITLE
Incorrect parsing of custom commands in citation suggestions

### ISSUE THREAD

**Issue by bilelomrani1** [James-Yu/LaTeX-Workshop#2096]
### Disable all the other extensions except for LaTeX Workshop, and check that you still see this issue. [Required]

You still see this issue?: Yes

## Describe the bug [Required]

IntelliSense fails to trigger correct autocomplete suggestion when custom macros are involved in the path argument of `\addbibresource`

### To Reproduce

1. Copy the following in a `bibliography.bib` file

```latex
@article{test,
	title = {Test title},
	author = {Last Name, First name},
}
```

2. Copy the following in a `main.tex` file

```latex
\providecommand{\main}{.}
\documentclass{article}
\usepackage{biblatex}

\addbibresource{\main/bibliography.bib}
\begin{document}

\cite{}

\end{document}
```
3. Query Intellisense autocompletion inside the braces of the `\cite` command (Alt+Space)

4. Observe no autocompletion

### Expected behavior

Intellisense shoud trigger citation suggestions in this case.

### LaTeX Workshop Output [Required]

```
[17:00:53] Formatting with command latexindent -c,/Users/bilelomrani/Desktop/test/,/Users/bilelomrani/Desktop/test/__latexindent_temp.tex,-y=defaultIndent: '    '
[17:00:53] Formatting failed with exit code 2
[17:00:53] stderr: Can't locate Log/Log4perl.pm in @INC (you may need to install the Log::Log4perl module) (@INC contains: /usr/local/texlive/2020/texmf-dist/scripts/latexindent /Library/Perl/5.18/darwin-thread-multi-2level /Library/Perl/5.18 /Network/Library/Perl/5.18/darwin-thread-multi-2level /Network/Library/Perl/5.18 /Library/Perl/Updates/5.18.4 /System/Library/Perl/5.18/darwin-thread-multi-2level /System/Library/Perl/5.18 /System/Library/Perl/Extras/5.18/darwin-thread-multi-2level /System/Library/Perl/Extras/5.18 .) at /usr/local/texlive/2020/texmf-dist/scripts/latexindent/LatexIndent/LogFile.pm line 22.
BEGIN failed--compilation aborted at /usr/local/texlive/2020/texmf-dist/scripts/latexindent/LatexIndent/LogFile.pm line 22.
Compilation failed in require at /usr/local/texlive/2020/texmf-dist/scripts/latexindent/LatexIndent/Document.pm line 25.
BEGIN failed--compilation aborted at /usr/local/texlive/2020/texmf-dist/scripts/latexindent/LatexIndent/Document.pm line 25.
Compilation failed in require at /Library/TeX/texbin/latexindent line 27.
BEGIN failed--compilation aborted at /Library/TeX/texbin/latexindent line 27.

[17:00:54] File watcher: responding to change in /Users/bilelomrani/Desktop/test/test.tex
[17:00:54] Parsing /Users/bilelomrani/Desktop/test/test.tex
[17:00:54] Cannot find .bib file \main/bibliography.bib
[17:00:54] Parse fls file.
```

## Desktop [Required]
 - OS: MacOSX 10.15.4
 - VS Code version: 1.45.1
 - Extension version: 8.9.0
 - TeX distribution version: TeX Live 2020
### Additional context

I use the custom macro `\main` in the path declaration of `bib` files because I use a subfile setup where I want to be able to compile subfiles separately. Because subfiles are located in subdirectories relative to the main file, I need to specify the path to the root of the project to correctly locate the `.bib` files.

**Comment by jlelong** [comment]
This is related to #1194. Unfortunately, expanding LaTeX macros when parsing the `.tex` files goes far beyond what we can achieve. instead we implemented in #1199 an alternative mechanism to detect missing dependencies from the last compilation.

To put the story short, you need to compile tour file once with `latexmk` to make all citations available through intellisense.

Btw, you seem to have an issue with `latexindent`, which complains about missing Perl modules. You should either fix your installation or disable automatic formatting.

**Comment by bilelomrani1** [comment]
Thank you for the explanation and sorry if the issue was unnecessary. I'll look for another solution then.
